src/main/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineFileType.java

```diff
@@ -0,0 +1,115 @@
+    XML(
+            List.of(".camelk.xml"),
+            "<!-- camel-k:",
+            CamelKModelineFileType::textIsFullyCommentedXML,
+            "<!-- camel-k: -->",
+            "Read more: https://camel.apache.org/camel-k/1.9.x/cli/modeline.html"),
+    Java(
+            List.of(".java"),
+            "// camel-k:",
+            CamelKModelineFileType::textIsFullyCommentedJava,
+            "// camel-k: ",
+            "Read more: https://camel.apache.org/camel-k/1.9.x/cli/modeline.html"),
+    YAML(
+            List.of(".camelk.yaml",".camelk.yml"),
+            "# camel-k:",
+            CamelKModelineFileType::textIsFullyCommentedYAML,
+            "# camel-k: ",
+            "Read more: https://camel.apache.org/camel-k/1.9.x/cli/modeline.html");
+
+    public final List<String> correspondingExtensions;
+    public final String modeline;
+    public final Function<String, Boolean> checkTextIsCommentsDelegate;
+
+    public final CompletionItem completion;
+
+    CamelKModelineFileType(List<String> correspondingExtensions,
+                           String modeline,
+                           Function<String, Boolean> checkTextIsCommentsDelegate,
+                           String completionLabel,
+                           String completionDocumentation) {
+        this.correspondingExtensions = correspondingExtensions;
+        this.modeline = modeline;
+        this.checkTextIsCommentsDelegate = checkTextIsCommentsDelegate;
+        this.completion = getCompletionItem(completionLabel, completionDocumentation);
+    }
+
+    public static CompletionItem getCompletionItem(String label, String documentation) {
+        CompletionItem completion = new CompletionItem(label);
+        completion.setDocumentation(documentation);
+
+        return completion;
+    }
+
+    public static Optional<CamelKModelineFileType> getFileTypeCorrespondingToUri(String uri) {
+        return List.of(CamelKModelineFileType.values()).stream()
+                .filter(type ->
+                        type.correspondingExtensions.stream().anyMatch(uri::endsWith)
+                )
+                .findFirst();
+    }
+
+    private static boolean textIsFullyCommentedXML(String text){
+        //Remove all segments between <!-- and -->. Check if it's empty.
+        Pattern commentRegex = Pattern.compile("<!--.*-->");
+
+        return textIsFullOfRegex(text, commentRegex);
+    }
+
+    private static boolean textIsFullyCommentedYAML(String text){
+        //Remove all segments between # and \n
+        Pattern commentRegex = Pattern.compile("#.*\\n");
+
+        return textIsFullOfRegex(text, commentRegex);
+    }
+
+    private static boolean textIsFullyCommentedJava(String text){
+        //Line Comments: Remove from // to \n
+        //Block Comments: Remove from /* to */. Newlines have to be explicitly added
+        Pattern lineComment = Pattern.compile("\\/\\/.*\\n");
+        Pattern blockComment = Pattern.compile("\\/\\*(.|\\n)*\\*\\/");
```

Comment:
Sonar is reporting a potential stackoverflow for this pattern.
https://rules.sonarsource.com/java/RSPEC-5998


if I understand it correctly, this shoudl be replcaed with `\\/\\*(?s).*\\*\\/`. Can I let you check it please?